### ISSUE TITLE
Inpainting edege softening support in background replacement with inpainting

### ISSUE THREAD

**Issue by lanyusan** [lllyasviel/Fooocus#1848]
**Read Troubleshoot**

[x] I admit that I have read the [Troubleshoot](https://github.com/lllyasviel/Fooocus/blob/main/troubleshoot.md) before making this issue.

I was trying to use Fooocus' excellent inpainting for product poster generation after the new Fooocus version with mask uploading support was published.

Fooocus' inpainting is really powerful combined with img2img support.

However there is an issue. The mask is binary so original anti aliasing or smoothing pixels at the  product object edge would be lost.

Then the new image generated with Fooocus with the mask would have sharp edge. I am wondering if there is any chance Fooocus could be enhanced to add inpainting edge softening or other techniques to address the problem.

I have attached some examples. Again thanks a lot for making such a wonderful tool.

![image](https://github.com/lllyasviel/Fooocus/assets/56706512/1ff87bc9-ff67-4545-943b-41dc4fa2edcc)

**Comment by mashb1t** [comment]
@lanyusan great description, thank you for the effort!
This is related to / a duplicate of https://github.com/lllyasviel/Fooocus/issues/1788, but with the addition of a mask. Will reference your mask proposal there.

**Comment by thiner** [comment]
Wow! this is great example of the usage of inpaint mask. 
But I think @mashb1t you misunderstood his question, he was trying to ask how to soften the edge. I think you should reopen this issue.

**Comment by mashb1t** [comment]
@thiner I still don't understand why this should differ from the other issue. Can you maybe rephrase how you understood the need here anc what die differences are to #1778? That would be great, thanks!

**Comment by thiner** [comment]
@mashb1t Ah, I got your point now. I thought you close this issue because of the "soften edges" requirement is implemented. In fact, you just guided author to watch the other relevant issue. Sorry for my misunderstanding. 😃
